### DISCUSSION TITLE
Headless Hosted Version

### DISCUSSION THREAD

**Discussion by TechStudent10** [ThioJoe/YT-Spammer-Purge#205] [Ideas]
## The Idea
What I am proposing is that instead of having people run the program on their computer, you could make it so that people can go to a website, enter their channel ID, authenticate with Google and store the channel ID in a database. Then, using a GitHub action, it would send a request to an API that will go through every channel and delete spam comments from their most recent video after a certain interval. Maybe like every two hours. Or, if the user doesn't want to authenticate with Google, they could set to only report the comments to YouTube. 

## The Problem
This would require you to store your `client_secrets.json` file on a server. This is very insecure because if a hacker gained access to the server, they could grab the `client_secrets.json` file and use it for malicious purposes.

## The Solution to the Problem
Pick a secure provider like AWS or Cloudflare. I'm pretty sure they have a low risk of getting hacked, however, I recommend doing research for what the most secure provider is.

**Comment by somebadcode** [comment]
Why would you use GitHub actions? Why not run it in GCP or similar and have it do the work at certain intervals? The current state of the script makes this approach very complicated.

**Reply by TechStudent10** [reply]
That is just a suggestion. It's open for debate whether to use GH Actions or not.
